src/components/swap/PendingModalContent/TransitionText.tsx

```diff
@@ -0,0 +1,81 @@
+import { Trans } from '@lingui/macro'
+import { useUnmountingAnimation } from 'hooks/useUnmountingAnimation'
+import { ReactNode, useEffect, useRef, useState } from 'react'
+import styled from 'styled-components'
+
+import { slideInAnimation, slideOutAnimation } from './animations'
+import { AnimationType } from './Logos'
+
+interface TransitionTextProps {
+  initialText: ReactNode
+  transitionText: ReactNode
+  transitionTimeMs?: number
+  onTransition?: () => void
+}
+
+const Container = styled.div`
+  position: relative;
+  width: 100%;
+  min-height: 30px;
+`
+
+const InitialTextContainer = styled.div`
+  width: 100%;
+  height: 100%;
+  position: absolute;
+  transition: display ${({ theme }) => `${theme.transition.duration.fast} ${theme.transition.timing.inOut}`};
+  ${slideInAnimation}
+  &.${AnimationType.EXITING} {
+    ${slideOutAnimation}
+  }
+`
+
+const TransitionTextContainer = styled.div`
+  width: 100%;
+  height: 100%;
+  position: absolute;
+  transition: display ${({ theme }) => `${theme.transition.duration.fast} ${theme.transition.timing.inOut}`};
+  ${slideInAnimation}
+  &.${AnimationType.EXITING} {
+    ${slideOutAnimation}
+  }
+`
+
+export function TransitionText({
```

Comment:
oh nice shared component here! maybe move this to `src/components/TransitionText` so it can be used in the future? nbd tho